lib/salesforce/donation_validator.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'salesforce/result'
+
+module Salesforce
+  class DonationValidator
+    def self.execute(data, supporter)
+      new(data, supporter).execute
+    end
+
+    def initialize(data, supporter)
+      @data = data
+      @supporter = supporter
+    end
+
+    def execute
+      Result.new(fields, errors)
+    end
+
+    private
+
+    attr_reader :data, :supporter
+
+    def fields
+      return unless errors.empty?
+      {
+        Amount: data.amount.to_s,
+        CloseDate: '2017-09-11',
+        Name: 'Online donation',
+        StageName: 'Received',
+        AccountId: supporter[:AccountId]
+      }
+    end
+
+    def errors
```

Comment:
`[validate_data, validate_amount, validate_account_id].compact`

Reply:
it looks like, because there are only three, doesn't it? But one-liners are not always more readable! I actually prefer it to look more like a list of tasks than like an array of errors, because it being an array of errors is more important from the point of view of the consumers of the class than for the validator class itself. In other words, this syntax tells the story better, in my opinion.